contributingGuides/review/RULES.md

```diff
@@ -0,0 +1,203 @@
+- **Severity**: SHOULD
+- **Condition**: When performing repeated lookups or checking existence in collections, convert arrays to Sets for O(1) performance instead of using O(n) methods like `.includes()`.
+
+Good:
+```tsx
+const reportIdFromRoute = report.reportId;
+const reportUserIdSet = new Set(Object.values(reports));
+const reportExists = reportUserIdSet.has(reportIdFromRoute);
+```
+
+Bad:
+```tsx
+const reportIdFromRoute = report.reportId;
+const reportExists = Object.values(reports).includes(reportIdFromRoute);
+```
+
+---
+
+### [PERF-3] Use early returns in array iteration methods
+- **Severity**: SHOULD
+- **Condition**: When using `.every()`, `.some()`, or similar methods, perform simple checks first with early returns before expensive operations.
+
+Good:
+```tsx
+const areAllTransactionsValid = transactions.every((transaction) => {
+    if (!transaction.rawData || transaction.amount <= 0) return false;
+    const validation = validateTransaction(transaction);
+    return validation.isValid;
+});
+```
+
+Bad:
+```tsx
+const areAllTransactionsValid = transactions.every((transaction) => {
+    const validation = validateTransaction(transaction);
+    return validation.isValid;
+});
+```
+
+---
+
+### [PERF-4] Use OnyxListItemProvider hooks instead of useOnyx in renderItem
+- **Severity**: MUST
+- **Condition**: Components rendered inside `renderItem` functions should use dedicated hooks from `OnyxListItemProvider` instead of individual `useOnyx` calls.
+
+Good:
+```tsx
+// ReportActionItem.tsx
+const personalDetails = usePersonalDetails();
+```
+
+Bad:
+```tsx
+// ReportActionItem.tsx
+const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST);
+```
+
+---
+
+### [PERF-5] Memoize objects and functions passed as props
+- **Severity**: MUST
+- **Condition**: Objects and functions passed as props should be properly memoized or simplified to primitive values to prevent unnecessary re-renders.
+
+Good:
+```tsx
+const reportData = useMemo(() => ({
+    reportID: report.reportID,
+    type: report.type,
+    isPinned: report.isPinned,
+}), [report.reportID, report.type, report.isPinned]);
+
+return <ReportActionItem report={reportData} />
+```
+
+Bad:
+```tsx
+const [report] = useOnyx(`ONYXKEYS.COLLECTION.REPORT${iouReport.id}`);
+
+return <ReportActionItem report={report} />
+```
+
+---
+
+### [PERF-6] Avoid inline object/array/function creation in JSX
+- **Severity**: SHOULD
+- **Condition**: Objects, arrays, and functions should not be created inline in JSX. Use `useMemo` or `useCallback` to memoize them when there's actual performance benefit.
+
+Good:
+```tsx
+const reportActionItemStyle = useMemo(() => [styles.container, styles.flex], []);
+const handleSelect = useCallback(() => {
+    onSelectRow(item);
+}, [onSelectRow, item]);
+
+<ReportActionItem
+    style={reportActionItemStyle}
+    onSelect={handleSelect}
+    reportID={report.reportID}
+    isPinned={report.isPinned}
```

Comment:
Remove this or add it to the bad example too

Reply:
I think the entire rule could be just https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md (is it not already?)